sklearn/utils/tests/test_param_validation.py

```diff
@@ -633,3 +619,22 @@ def test_cv_objects():
     assert constraint.is_satisfied_by([([1, 2], [3, 4]), ([3, 4], [1, 2])])
     assert constraint.is_satisfied_by(None)
     assert not constraint.is_satisfied_by("not a CV object")
+
+
+def test_third_party_estimator():
+    """Check that the validation from a scikit-learn estimator inherited by a third
+    party estimator does not impose a match between the dict of constraints and the
+    parameters of the estimator.
+    """
+
+    class ThirdPartyEstimator(_Estimator):
+        def __init__(self, b):
+            self.b = b
+            super().__init__(a=0)
+
+        def fit(self, X=None, y=None):
+            super().fit(X, y)
+
+    # does not raise, niether because "b" is not in the constraints dict, neither
+    # because "a" is not a parameter of the estimator.
```

Comment:
Nit: I think this is a little easier to parse:

```suggestion
    # does not raise, even tho "b" is not in the constraints dict and
    # "a" is not a parameter of the estimator.
```

Reply:
Alternatively:
``` suggestion
    # Does not raise an error, despite the fact that:
    # 1) "a" is not a parameter of the estimator
    # 2) "b" is not present in the constraints dict
```